addons/binding/org.openhab.binding.pidcontroller/README.md

```diff
@@ -0,0 +1,97 @@
+# PID Controller Binding
+
+The PID Controller binding provides PID controllers with the purpose of fine tuning systems using input/setpoint variables (such as temperature control).
+For a glimpse of how a PID Controller works, please check: https://en.wikipedia.org/wiki/PID_controller 
+
+The binding provides 9 channels,  3 (input, setpoint, output) of them being the required channels for the binding to cause result for the items linked to the channels. The other 6 (LoopTime, kpadjuster, kiadjuster, kdadjuster, pidlowerlimit, pidupperlimit) channels (configuration channels for the controller) only provide tweaking  of the controllers initialized, if not set the controller defaults to the following:
+
+LoopTime=1000 (milliseconds)
+
+kpadjuster=1 (multiplier for the Kp - proportional constant in the PID controller)
+
+kiadjuster=1 (multiplier for the Ki - integral constant in the PID controller)
+
+kdadjuster=1 (multiplier for the Kd - derivative constant in the PID controller)
+
+pidlowerlimit=-255 (the lower limit of the output of controller)
+
+pidupperlimit=255 (the upper limit of the output of controller)
+
+In order for the binding to produce result, the input, setpoint and output channels have to be linked to items. 
+
+## Binding configuration
+
+The binding does not need any configuration, all the configuration options of the controllers are available through channels.
+
+##Thing configuration
+
+In the things folder, create a file called pidcontroller.things (or any other name) and configure your controllers inside.
+
+Example:
+
+```
+pidcontroller.Controller.Your_Controller_Name
+
+```
+
+
+## Channels
+
+All channels are provided as Number type.
+*input: the input of the controller (from a sensor)
+*setpoint: the setpoint of the controller (a virtual item or an item receiving setpoint status from the outside world)
+*output: the output of the controller

```

Comment:
The output is by default in the form of 1 byte (0-255) so that it can be sent to the actuators that are affecting the measurement value. Through the channels PIDUpperLimit and PIDLowerLimit, these can be modified. The idea is to have the scale the user desires.
 Example: constant lighting application where you measure the light level, set the required level and adjust the dimmable lights accordingly. Without a PID control you would have an oscillating system, or even worse an unstable system.
The temperature application is also one of them.